plutus-core/prelude/PlutusPrelude.hs

```diff
@@ -159,6 +161,12 @@ foldMapM f xs = foldr step return xs mempty where
 reoption :: (Foldable f, Alternative g) => f a -> g a
 reoption = foldr (const . pure) empty
 
+enumeration :: (Bounded a, Enum a) => [a]
+enumeration = [minBound .. maxBound]
+
+tabulate :: (Bounded a, Enum a, Ix a) => (a -> b) -> Array a b
+tabulate f = listArray (minBound, maxBound) $ map f enumeration
```

Comment:
Nit: I associate this name with [tabulate](http://hackage.haskell.org/package/representable-functors-3.2.0.2/docs/Data-Functor-Representable.html)

Reply:
Exactly. I just didn't bother creating an instance (like people [do for `Vector`s](https://hackage.haskell.org/package/total-maps-1.0.0.2/docs/src/Data-Total-Array.html#TotalArray)). Maybe should've called it `tabulateArray` or something.

Reply:
No, I mean I didn't want to add that package to dependencies just to provide an instance instead of using the function directly. I believe, the instance should be definable.

> I wonder why there isn't an instance of `Representable`

Probably just an overlook.

Reply:
So I looked into it again and while implementing `Representable.tabulate` for `Array` seems perfectly reasonable, there's that:

```haskell
class (Distributive f, Indexable f) => Representable f
```

and `Array i` is not `Distributive` (unless we `newtype` it to ensure that it has mappings over the whole range of `i`).

So yeah, I'll just rename the function to `tabulateArray` and add a comment.